rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/InjectionUtils.java

```diff
@@ -1167,10 +1161,12 @@ public static void injectContextMethods(Object requestObject,
                 if (!cri.isSingleton()) {
                     InjectionUtils.injectThroughMethod(requestObject, method, o, message);
                 } else {
-                    ThreadLocalProxy<Object> proxy 
-                        = (ThreadLocalProxy<Object>)cri.getContextSetterProxy(method);
+                    Object proxy = extractFromSetter(requestObject, method);
+                    if (!(proxy instanceof ThreadLocalProxy)) {
```

Comment:
Hi! Sorry for the late response!

The intention of this condition is to prevent NPE or ClassCastException that may happen in case `extractFromSetter()` fails or in case it returns some completely unexpected value.
The unexpected value could happen from the unfortunate coincidence in which a getter exists but it doesn't follow a Bean Property pattern. 
Another possibility would be the property being overwritten by code outside cxf (That should be unlikely, but it is possible) and therefore no longer a proxy.